crates/astria-bridge-withdrawer/src/bridge_withdrawer/startup.rs

```diff
@@ -0,0 +1,613 @@
+    ///    sequencer.
+    /// 2. The last transaction by the bridge account failed to execute (this should not happen in
+    ///    the sequencer logic)
+    /// 3. The last transaction by the bridge account did not contain a withdrawal action
+    /// 4. The memo of the last transaction by the bridge account could not be parsed
+    async fn get_starting_rollup_height(&mut self) -> eyre::Result<u64> {
+        let signed_transaction = self
+            .get_last_transaction()
+            .await
+            .wrap_err("failed to get the bridge account's last sequencer transaction")?;
+        let starting_rollup_height = if let Some(signed_transaction) = signed_transaction {
+            rollup_height_from_signed_transaction(&signed_transaction)
+                .wrap_err(
+                    "failed to extract rollup height from last transaction by the bridge account",
+                )?
+                .checked_add(1)
+                .wrap_err("failed to increment rollup height by 1")?
+        } else {
+            1
+        };
+        Ok(starting_rollup_height)
+    }
+}
+
+/// Extracts the rollup height from the last transaction by the bridge account on the sequencer.
+/// Since all the withdrawals from a rollup block are batched into a single sequencer transaction,
+/// he rollup height can be extracted from the memo of any withdrawal action in the batch.
+///
+/// # Returns
+///
+/// The rollup height of the last batch of withdrawals.
+///
+/// # Errors
+///
+/// 1. The last transaction by the bridge account did not contain a withdrawal action.
+/// 2. The memo of the last transaction by the bridge account could not be parsed.
+/// 3. The block number in the memo of the last transaction by the bridge account could not be
+///   converted to a u64.
+fn rollup_height_from_signed_transaction(
+    signed_transaction: &SignedTransaction,
+) -> eyre::Result<u64> {
+    // find the last batch's rollup block height
+    let withdrawal_action = signed_transaction
+        .actions()
+        .iter()
+        .find(|action| matches!(action, Action::BridgeUnlock(_) | Action::Ics20Withdrawal(_)))
+        .ok_or_eyre("last transaction by the bridge account did not contain a withdrawal action")?;
+
+    let last_batch_rollup_height = match withdrawal_action {
+        Action::BridgeUnlock(action) => {
+            let memo: BridgeUnlockMemo = serde_json::from_slice(&action.memo)
+                .wrap_err("failed to parse memo from last transaction by the bridge account")?;
+            Some(memo.block_number.as_u64())
+        }
+        Action::Ics20Withdrawal(action) => {
+            let memo: Ics20WithdrawalFromRollupMemo = serde_json::from_str(&action.memo)
+                .wrap_err("failed to parse memo from last transaction by the bridge account")?;
+            Some(memo.block_number)
+        }
+        _ => None,
+    }
+    .expect("action is already checked to be either BridgeUnlock or Ics20Withdrawal");
```

Comment:
This is kind of far away from where the actual check happens. Easy to accidentally mess up when refactoring. Please just `bail!` here instead, I don't think one needs to panic.